temporalio/lib/temporalio/worker.rb

```diff
@@ -484,13 +486,35 @@ def initialize(
 
       # Validate worker
       @bridge_worker.validate
+
+      # Mutex needed for accessing and replacing a client
+      @client_mutex = Mutex.new
     end
 
     # @return [String] Task queue set on the worker options.
     def task_queue
       @options.task_queue
     end
 
+    # @return [Client] Client for this worker. This is the same as {Options.client} in {options}, but surrounded by a
+    #   mutex to be safe for client replacement in {client=}.
+    def client
+      @client_mutex.synchronize { @options.client }
```

Comment:
Do we really need to acquire the mutex to read the client? Even during client replacement, that prop would be, at any moment, either the previous client or the new client, which are both valid values.